java/flight/src/main/java/org/apache/arrow/flight/FlightService.java

```diff
@@ -191,17 +191,17 @@ public void completed() {
             StreamPipe.wrap(responseObserver, PutResult::toProtocol)).run();
         responseObserver.onCompleted();
       } catch (Exception ex) {
-        logger.error("Failed to process custom put.", ex);
         responseObserver.onError(ex);
-        // The client may have terminated, so the exception here is effectively swallowed.
         // Log the error as well so -something- makes it to the developer.
         logger.error("Exception handling DoPut", ex);
-      }
-      try {
-        fs.close();
-      } catch (Exception e) {
-        logger.error("Exception closing Flight stream", e);
-        throw new RuntimeException(e);
+        throw new RuntimeException(ex);
```

Comment:
What will actually handle or see this exception? gRPC should only need the `onError` call above.

Reply:
Good question.
In addition to process the exception by onError, it is also desirable to process the exception by the Arrow framework, because that will make problem diagnosis much easier.

Throwing an exception instead of swallowing it is the first step towards this goal.